packages/collector/test/tracing/opentelemetry/test.js

```diff
@@ -516,7 +520,8 @@ mochaSuiteFn('opentelemetry/instrumentations', function () {
           }
         });
 
-        await controls.startAndWaitForAgentConnection();
+        // Waiting time added for the agent connection
+        await controls.startAndWaitForAgentConnection(5000, Date.now() + 1000 * 60 * 2);
```

Comment:
Putting it here.

Q: How is the solution in the PR working if the tedious library does not fire the `connection.on('connect', err => {` event? 🤔

Reply:
My understanding is that the library is firing the event but not handling the response when the database status is paused. It was waiting for the connection timeout. But our tests were failing before that because the test cases were not waiting.

Reply:
I'd suggest lets wait exactly 24h and not running the tests. And then manually run the tests locally.
If it passes one the first run, then we are good 👍

Reply:
Okay sure. I will check and post the results by tomorrow 👍

Reply:
@kirrg001 The tests are running perfectly even after 24h

Reply:
Perfect 👍